packages/react-topology/src/components/TopologyView/examples/TopologyContextMenuDemo.tsx

```diff
@@ -0,0 +1,201 @@
+import * as React from 'react';
+
+// eslint-disable-next-line patternfly-react/import-tokens-icons
+import { RegionsIcon as Icon1 } from '@patternfly/react-icons';
+
+import {
+  ColaLayout,
+  ComponentFactory,
+  DefaultEdge,
+  DefaultGroup,
+  DefaultNode,
+  EdgeStyle,
+  Graph,
+  GraphComponent,
+  Layout,
+  LayoutFactory,
+  Model,
+  ModelKind,
+  Node,
+  NodeModel,
+  NodeShape,
+  SELECTION_EVENT,
+  Visualization,
+  VisualizationProvider,
+  VisualizationSurface,
+  withContextMenu,
+  WithContextMenuProps,
+  ContextMenuSeparator,
+  ContextMenuItem
+} from '@patternfly/react-topology';
+
+interface CustomNodeProps {
+  element: Node;
+}
+
+const CustomNode: React.FC<CustomNodeProps & WithContextMenuProps> = ({ element, onContextMenu, contextMenuOpen }) => {
+  const Icon = Icon1;
+
+  return (
+    <DefaultNode element={element} onContextMenu={onContextMenu} contextMenuOpen={contextMenuOpen}>
+      <g transform={`translate(25, 25)`}>
+        <Icon style={{ color: '#393F44' }} width={25} height={25} />
+      </g>
+    </DefaultNode>
+  );
+};
+
+const customLayoutFactory: LayoutFactory = (type: string, graph: Graph): Layout | undefined => {
+  switch (type) {
+    case 'Cola':
+      return new ColaLayout(graph);
+    default:
+      return new ColaLayout(graph, { layoutOnDrag: false });
+  }
+};
+
+const customComponentFactory: ComponentFactory = (kind: ModelKind, type: string) => {
+  switch (type) {
+    case 'group':
+      return DefaultGroup;
+    default:
+      switch (kind) {
+        case ModelKind.graph:
+          return GraphComponent;
+        case ModelKind.node:
+          return withContextMenu(() => contextMenu)(CustomNode);
```

Comment:
Let's use a function here to create the context menu rather than returning a static context menu. Also, let's add the context menu to the graph had edges as well.